src/node/internal/crypto_spkac.ts

```diff
@@ -0,0 +1,103 @@
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+/* todo: the following is adopted code, enabling linting one day */
+/* eslint-disable */
+
+import { default as cryptoImpl } from 'node-internal:crypto';
+import {
+  StringLike,
+  Buffer,
+} from 'node-internal:internal_buffer';
+
+import {
+  isArrayBufferView,
+  isAnyArrayBuffer,
+} from 'node-internal:internal_types';
+
+import {
+  ERR_INVALID_ARG_TYPE,
+} from 'node-internal:internal_errors'
+
+type ArrayLike = cryptoImpl.ArrayLike;
+
+function getArrayBufferOrView(buffer: unknown, name: string, encoding?: string) {
+  if (isAnyArrayBuffer(buffer))
+    return buffer;
+  if (typeof buffer === 'string') {
+    if (encoding === 'buffer')
+      encoding = 'utf8';
+    return Buffer.from(buffer, encoding);
+  }
+  if (!isArrayBufferView(buffer)) {
+    throw new ERR_INVALID_ARG_TYPE(
+      name,
+      [
+        'string',
+        'ArrayBuffer',
+        'Buffer',
+        'TypedArray',
+        'DataView',
+      ],
+      buffer,
+    );
+  }
+  return buffer;
+}
+
+export function verifySpkac(spkac: StringLike | ArrayLike, encoding?: string) {
+  return cryptoImpl.verifySpkac(getArrayBufferOrView(spkac as unknown, 'spkac', encoding));
+}
+
+export function exportPublicKey(spkac: StringLike | ArrayLike, encoding?: string) {
+  const ret = cryptoImpl.exportPublicKey(getArrayBufferOrView(spkac as unknown, 'spkac', encoding));
+  return ret ? Buffer.from(ret) : Buffer.alloc(0);
+}
+
+export function exportChallenge(spkac: StringLike | ArrayLike, encoding?: string) {
+  const ret = cryptoImpl.exportChallenge(getArrayBufferOrView(spkac as unknown, 'spkac', encoding));
+  return ret ? Buffer.from(ret) : Buffer.alloc(0);
+}
+
+// The legacy implementation of this exposed the Certificate
+// object and required that users create an instance before
+// calling the member methods. This API pattern has been
+// deprecated, however, as the method implementations do not
+// rely on any object state.
+
+// For backwards compatibility reasons, this cannot be converted into a
+// ES6 Class.
+export function Certificate(this: any) {
+  if (!(this instanceof Certificate))
+    return new (Certificate as any)();
+}
+
+Certificate.prototype.verifySpkac = verifySpkac;
```

Comment:
Since the non-static methods of Certificate are deprecated, do we need to define them?

Reply:
I've still seen the deprecated/legacy methods used in the wild. Likely a good idea to retain them here.

Reply:
Ok, in this case it makes sense since the deprecated methods look sane and there's little overhead in having them.